crates/task-impls/src/transactions.rs

```diff
@@ -120,9 +120,245 @@ pub struct TransactionTaskState<TYPES: NodeType, I: NodeImplementation<TYPES>> {
+        let version = match hotshot_types::simple_certificate::version(
+            block_view,
+            &self
+                .decided_upgrade_certificate
+                .read()
+                .await
+                .as_ref()
+                .cloned(),
+        ) {
+            Ok(v) => v,
+            Err(err) => {
+                error!("Upgrade certificate requires unsupported version, refusing to request blocks: {}", err);
+                return None;
+            }
+        };
+
+        // Only request bundles and propose with a nonempty block if we are not between versions.
+        if !self
+            .decided_upgrade_certificate
+            .read()
+            .await
+            .as_ref()
+            .is_some_and(|cert| cert.upgrading_in(block_view))
+        {
+            if let Ok(bundles) = self
+                .auction_results_provider
+                .fetch_bundles(block_view)
+                .await
+            {
+                let Some(null_fee) =
+                    null_block::builder_fee(self.membership.total_nodes(), version)
+                else {
+                    error!("Failed to get null fee");
+                    return None;
+                };
+
+                let mut sequencing_fees = Vec::new();
+                let mut bid_fees: Vec<BuilderFee<TYPES>> = Vec::new();
+                let mut transactions: Vec<
+                    <TYPES::BlockPayload as BlockPayload<TYPES>>::Transaction,
+                > = Vec::new();
+
+                for bundle in bundles {
+                    sequencing_fees.push(bundle.sequencing_fee);
+                    bid_fees.push(null_fee.clone());
+                    transactions.extend(bundle.transactions);
+                }
+
+                let validated_state = self.consensus.read().await.decided_state();
+
+                if let (Ok(sequencing_fees), Ok(bid_fees), Ok((block_payload, metadata))) = (
+                    Vec1::try_from_vec(sequencing_fees),
+                    Vec1::try_from_vec(bid_fees),
+                    TYPES::BlockPayload::from_transactions(
+                        transactions,
+                        &validated_state,
+                        &self.instance_state.clone(),
+                    )
+                    .await,
+                ) {
+                    broadcast_event(
+                        Arc::new(HotShotEvent::BlockRecv(PackedBundle::new(
+                            block_payload.encode(),
+                            metadata,
+                            block_view,
+                            bid_fees,
+                            sequencing_fees,
+                            None,
+                        ))),
+                        &event_stream,
+                    )
+                    .await;
+
+                    return None;
+                }
+            }
+        }
+
+        // If we couldn't get any bundles, send an empty block
+        warn!(
```

Comment:
Another failure case is when the solver never returns any results.  That should also cause an empty block to be proposed.

Reply:
Clarified the comment -- this was already the case in the logic!